### ISSUE TITLE
Feature request: allow destructuring the scopedSlot scope variable

### ISSUE THREAD

**Issue by miljan-aleksic** [vuejs/vue#4791]
I don't know if the implementation is simple, but the example it is. The scope variable could have many properties but when rendering usually you just need a few. 

```vue
// Currently
<template slot="cell" scope="props">
  {{ props.row }}
</template>

// Ideally
<template slot="cell" scope="{{ row }}">
  {{ row }}
</template>
````

**Comment by LinusBorg** [comment]
If I'm not mistaken, that is actually already possible when using vue-loader?

**Comment by simplesmiler** [comment]
Yes, [ES2015 in templates](https://github.com/vuejs/vue-loader/blob/v10.0.0/docs/en/features/es2015.md#es2015-in-templates).

**Comment by miljan-aleksic** [comment]
Is not working in my project, using vue-loader 10.0.3. In the same component I am using ES2015 code which works fine, but for this specific case its rising syntax errors. Tried as well `<template slot="cell" :scope="{{ row }}">`.

**Comment by LinusBorg** [comment]
not `{{props}}` - it's `{ props }`

[ES2015 object destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Object_destructuring)

**Comment by miljan-aleksic** [comment]
facepalm.... I don't know where my head is... Sorry guys :) and thanks.

**Comment by jronwesternvp** [comment]
This works fine, but assigning to new variable names seems to not work.

 ... slot-scope="{ newVariable: destructuredVariable }" ...

Can anyone check it out? Thanks

**Comment by LinusBorg** [comment]
No, we won't. 

please open a big report with a reproduction.

**Comment by emahuni** [comment]
I wish this was in the Vue documentation under scoped slots with an example also so that everybody could see that it's possible and how this is done. I saw this in another [component](https://vue-apollo.netlify.com/guide/components/query.html#query-gql-tag) being used and though I understood what was happening I didn't remember seeing it in the Vue documentation and  wondered why it wasn't in there. looking for the syntax I ended up here. This actually made clearer on what was happening , which means that it'd be great if put in the docs.
